src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -183,7 +188,9 @@ public void visitToken(final DetailAST ast) {
                 break;
             case TokenTypes.OBJBLOCK:
                 final int parentType = ast.getParent().getType();
```

Comment:
Please move this code to a helper method. Something like `boolean isClassOrEnumOrRecordDef(DetailAST ast)`
This triple check is performed several times in the file.

Reply:
Since we check the parent twice, and the token itself once, I left the code that is extracting the parent type in each case group.  Also, since there was already a method called `isClassOrEnumOrRecordDef` used for checking and setting frames, I named the new method `astTypeIsClassOrEnumOrRecordDef`.